Framework/Geometry/src/Instrument/ComponentInfo.cpp

```diff
@@ -457,32 +392,25 @@ BoundingBox ComponentInfo::boundingBox(const size_t componentIndex,
       componentType(componentIndex) == Beamline::ComponentType::Infinite) {
     return componentBoundingBox(componentIndex, reference);
   }
+
   BoundingBox absoluteBB;
-  auto rangeComp = m_componentInfo->componentRangeInSubtree(componentIndex);
-  std::map<size_t, size_t> detExclusions{};
-  auto compIterator = rangeComp.rbegin();
-  while (compIterator != rangeComp.rend()) {
-    const size_t index = *compIterator;
-    const auto compFlag = componentType(index);
-    if (hasSource() && index == source()) {
-      ++compIterator;
-    } else if (compFlag == Beamline::ComponentType::Rectangular ||
-               compFlag == Beamline::ComponentType::Structured) {
-      growBoundingBoxAsRectuangularBank(index, reference, absoluteBB,
-                                        detExclusions, compIterator);
-    } else if (compFlag == Beamline::ComponentType::OutlineComposite) {
-      growBoundingBoxAsOutline(index, reference, absoluteBB, detExclusions,
-                               compIterator);
-    } else {
-      // General case
-      absoluteBB.grow(componentBoundingBox(index, reference));
-      ++compIterator;
+  const auto compFlag = componentType(componentIndex);
+  if (hasSource() && componentIndex == source()) {
+    // Do nothing. Source is not considered part of the beamline for bounding
+    // box calculations.
+  } else if (compFlag == Beamline::ComponentType::Unstructured) {
+    for (const auto &childIndex : this->children(componentIndex)) {
```

Comment:
Is my understanding correct here: This `for` loop is essentially replacing the old `while` that used to be outside the `if/else` block?